src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,62 @@
+package duke;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+
+/**
+ * Ui class handles the user interface of the application.
+ */
+public class Ui {
+    protected Scanner in;
```

Comment:
Perhaps can give a name like userInput, for example.
A name is not just for differentiation; it should explain the named entity to the reader accurately and at a sufficient level of detail. (can refer Code Quality: Naming)